src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp

```diff
@@ -339,6 +339,21 @@ void SyncInferRequest::wait() {
                         "than required (", output_memory->size(), ")");
 
         bool need_output_update = output_layout.bytes_count() == 0 || (output_memory && output_tensor->get_byte_size() != output_memory->size());
+
+        // Check shape is changed when size between output_tensor and ouytput_memory are same.
+        if (!need_output_update) {
+            auto output_layout_shape = output_layout.get_shape();
+            auto output_tensor_shape = output_tensor->get_shape();
+            if (!output_layout_shape.empty() && !output_tensor_shape.empty()) {
+                for (size_t i = 0; i < output_layout_shape.size(); i++) {
```

Comment:
nit: you can directly compare two shapes using == operator instead of iterating over all dimensions in the loop

Reply:
Updated it! Thanks!